frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx

```diff
@@ -66,42 +63,71 @@ const FieldsAndSoilComponent: React.FC<InputModuleProps> = ({ farmDetails, updat
       validationSchema={validationSchema}
       onSubmit={onSubmit}
     >
-      <Form>
-        <StyledFarmInfo>
-          <div id="inputContainer">
-            <CustomField
-              label="Year"
-              id="Year"
-              name="Year"
-              type="number"
-              width="20%"
-            />
-
-            <CustomField
-              label="Farm Name"
-              id="FarmName"
-              name="FarmName"
-              type="text"
-            />
-          </div>
-
-          <StyledSelectContainer>
-            <CustomSelect
-              name="FarmRegion"
-              id="FarmRegion"
-              label="Region"
-              options={options}
-              width="50%"
-            />
-            <Button
-              type="submit"
-              size="sm"
-              disabled={false}
-              text={ComponentText.NEXT}
-            />
-          </StyledSelectContainer>
-        </StyledFarmInfo>
-      </Form>
+      {({ values }) => (
+        <Form>
+          <StyledFarmInfo>
+            <div id="inputContainer">
+              <CustomField
+                label="Field name"
+                id="FieldName"
+                name="FieldName"
+                type="text"
+              />
+              <CustomField
+                label="Area"
+                id="Area"
+                name="Area"
+                type="number"
+                width="20%"
+              />
+            </div>
+            <StyledTextAreaContainer>
+              <CustomTextArea
+                name="Comments"
+                id="Comments"
+                label="Comments (optional)"
+                placeholder="e.g., poor drainage in southwest corner (no need to specify crop here)"
+                width="50%"
+              />
+              <StyledButtonGroupContainer>
+                <Button
+                  type="reset"
+                  size="sm"
+                  disabled={false}
+                  text={ComponentText.CANCEL}
+                />
+                <Button
+                  type="button"
+                  size="sm"
+                  disabled={false}
+                  text={ComponentText.ADD}
+                  handleClick={() => {
+                    const farmInfo: FarmDetailsInterface = { ...fieldsInfo };
+                    farmInfo.FieldName = values.FieldName;
+                    farmInfo.Area = values.Area;
+                    farmInfo.Comments = values.Comments;
+                    setFieldsInfo(farmInfo);
+                  }}
+                />
```

Comment:
This arrow function should be moved to the top of the component, in it's own function. 
It seems to be correct though.

Would `fieldInfo` be a better name for `farmInfo` though? I found it a bit too close to farmDetails, and this is only building info about the fields, right?

Later in implementation, on the final `next` button, you can call `updateFarmDetails` which is the `setFarmDetails` set function. It should be passed as props to this `FieldsAndSoil: InputModule` from MainPage. `updateFarmDetails` is what is used to set this new data object into our main data object, `farmDetails`.

Let me know if the explanation didn't make sense :stuck_out_tongue_winking_eye:

Reply:
FieldInfo is only building on the fields but later on updateFarmDetails gets called with the values that was inputted. I am really unsure with how this implemented, and I'm pretty sure there's a better way.
I feel that it's not DRY, as this function does the same thing as onSubmit without being a submit button and going through the next section.

Reply:
Also, I've tried creating a function at the top of component instead of inline. But once it's outside formik, values are undefined. I've used, useFormikContext<SubmissionValues> from Formik dependency to let me use the values that was used in Formik. But still returns undefined.

When I've searched around it should be inside Formik. We can create another component and let it be a child of the Component with the Formik.

FormComponent = ...
const formik = useFormikContext<SubmissionValues>();

<Formik
...
<Form Component>

If you get my point! But that's more lines tbh. So inline function was shorter for me.

Reply:
FormComponent = ...
const formik = useFormikContext<SubmissionValues>();

<Formik>
<Form Component>
</Formik>

If you get my point! But that's more lines tbh. So inline function was shorter for me.

Reply:
Was able to move it and make it work.